src/core/mount.c

```diff
@@ -1840,9 +1840,18 @@ static bool mount_is_mounted(Mount *m) {
 
 static int mount_on_ratelimit_expire(sd_event_source *s, void *userdata) {
         Manager *m = userdata;
+        Job *j;
 
         assert(m);
 
+        /* Let's enqueue all start jobs that were previously skipped because of active ratelimit. */
+        HASHMAP_FOREACH(j, m->jobs) {
+                if (j->unit->type != UNIT_MOUNT || j->type != JOB_START || !j->was_skipped)
+                        continue;
+
+                job_add_to_run_queue(j);
+        }
```

Comment:
I think we can safely dumb this down: simply enqueue any job for a mount unit again, regardless what it is. Adding something to the queue can basically be done on a hunch, there's no need to guarantee that anything really changed. Hence, I'd really enqueue any job for a mount unit without trying to be too smart here. It's not that this is going to be a million things, it's just a small number, and it makes things really robust.